File: src/pages/latest-news.py
# from flask import Flask, jsonify
# from flask_cors import CORS
# import requests

# app = Flask(__name__)
# CORS(app)  # Enable CORS for frontend access

# API_KEY = "9f98d50b90ed46d4bb17143bd35c6bb3"  # Replace with your actual NewsAPI key

# @app.route('/latest-news', methods=['GET'])
# def get_latest_news():
#     try:
#         url = f"https://newsapi.org/v2/top-headlines?country=in&apiKey={API_KEY}"
#         response = requests.get(url)

#         # Handle API request failure
#         if response.status_code != 200:
#             print("NewsAPI Error:", response.text)
#             return jsonify({"error": "Failed to fetch latest news"}), 500

#         data = response.json()

#         if "articles" in data and data["articles"]:
#             articles = [{"title": article.get("title", "No Title"), 
#                          "source": article.get("source", {}).get("name", "Unknown")} 
#                          for article in data["articles"][:10]]
#             return jsonify({"articles": articles})

#         return jsonify({"error": "No latest news found!"}), 404

#     except Exception as e:
#         print("Error in /latest-news:", str(e))
#         return jsonify({"error": "Internal Server Error"}), 500

# if __name__ == '__main__':
#     app.run(debug=True, port=5000)

File: src/pages/LatestNews.js
import React, { useState, useEffect } from "react";
import axios from "axios";
import "../styles/LatestNews.css";

const LatestNews = () => {
  const [latestNews, setLatestNews] = useState([]);
  const [error, setError] = useState("");
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    const fetchLatestNews = async () => {
      try {
        const response = await axios.get("http://127.0.0.1:5000/latest-news");
        setLatestNews(response.data.articles);
      } catch (err) {
        setError("Failed to load latest news. Try again later.");
      }
      setLoading(false);
    };

    fetchLatestNews();
  }, []);

  return (
    <div className="latest-news-container">
      <h2 className="news-title">🆕 Latest News</h2>
      {loading ? <p>Loading news...</p> : error ? <p>{error}</p> : (
        <ul className="news-list">
          {latestNews.map((news, index) => (
            <li key={index}>
              <strong>{news.source.name}:</strong> {news.title}
            </li>
          ))}
        </ul>
      )}
    </div>
  );
};

export default LatestNews;
